### PR TITLE
project configuration, tox, travis, codecov

### PR DESCRIPTION
**Pull request by eLRuLL**

### CODE REVIEW

**Comment by redapple** [comment]
@eLRuLL , thanks for this.
You can also update `setup.py` classifiers to state Python 3 support

File: tox.ini [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
-envlist = py27
+envlist = py27,py34,py35
 
 [testenv]
 deps =
     -rrequirements.txt
-    nose
+    pytest
+    pytest-cov
+    hypothesis
```
Comment (redapple) [inline tox.ini]: is this used?
(I'm not familiar with `hypothesis(-pytest)`)

**Comment by eLRuLL** [comment]
@redapple setup also updated 😄

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+    - TOXENV=py27
+    - TOXENV=py34
+    - TOXENV=py35
+
 install:
-  - pip install -U tox
+    - pip install -U tox codecov
+
 script: tox
-deploy:
```
Comment (redapple) [inline .travis.yml]: I missed that earlier.
why is the deploy section removed?

Comment (eLRuLL) [inline .travis.yml]: my bad, I added it back

**Comment by eLRuLL** [comment]
ready

File: .travis.yml [hunk 3]
```diff
@@ -1,11 +1,28 @@
...
 install:
-  - pip install -U tox
+    - pip install -U tox codecov
+
 script: tox
+
+after_success:
+    - codecov
+
+cache:
```
Comment (redapple) [inline .travis.yml]: is this section required?

Comment (eLRuLL) [inline .travis.yml]: only useful for faster builds, I see it being used on several SH projects.

**Comment by redapple** [comment]
LGTM

**Pull request merged**